### ISSUE TITLE
Circular references freeze browser on $save() or RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by hinok** [platanus/angular-restmod#223]
Hey, I found that cirucal references between models freeze browser after call $save() on a model.
It was quite hard to find because circucal dependecy was not so obvious at the beginning for me.

I've built very simple example which demonstrates this issue: http://plnkr.co/edit/DcsIHDxxlKwhQv8aF0KU?p=preview

It's weird because in my application I don't get `Maximum call stack size exceeded`, but on plnkr after few seconds I get it.

Is there any save way to define extra dynamic properties on instantiated model? By dynamic I mean properties not defined in restmod model definition. Maybe they could be prefixed in some way?

It's related to https://github.com/platanus/angular-restmod/issues/195 but blackjid's solution assumes that all properties are known even before use.

**Comment by iobaixas** [comment]
Hi Dawid,

The idea is to reference restmod objects from other restmod objects using relations, when doing so the relation builder takes care of properly handling the nested object serialization. Could you elaborate a little more in the use case so maybe I can suggest a better approach?

**Comment by hinok** [comment]
In a large nutshell
* I use `lodash` in my project
* my app allows for allocating tickets (of music festivals) to a person who can sell those tickets later
* I was iterating over collection of plain javascript objects and then over restmod collection
* I have two models: Event, Ticket (event has many tickets, ticket has one event)

In case when someone want allocate ticket: `Event X - Ticket Y` to a person, that person can have already allocated ticket `Event X - Ticket Z`. Because I display tickets grouped by events, I have to check whether user has any tickets from an event before I do any action. In case when event doesn't exist, I have to create/copy event object and push ticket.

I've used _.find() which performs a deep comparison of each element in a collection to the given properties object. So in my case when:

```
campaign has many tickets
ticket has one campaign
```

_.find iterates endlessly. It's quite obvious but I had that problem because at the beginning I've been iterating over collection of plain JS objects rather than restmod collection. When I've started iterating over restmod events collection, that small change as result freezes the browser. It was not so easy to spot.

I think that an error or warning could be helpful in that cases.
